forte/data/extractor/vocabulary.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import defaultdict
+from typing import Any
+
+
+class Vocabulary:
+    '''This class is used to save the mapping table from
+    tokens, words, labels etc. to indexes.
+    There are two methods used for the mapping table.
+    One is indexing, which will map the entry to a sigle interger.
+    The other one is one-hot, which will map the entry to an
+    one-hot vector
+    '''
+
+    PAD_ENTRY = "<PAD>"
+    UNK_ENTRY = "<UNK>"
+
+    def __init__(self, method, use_pad, use_unk):
+        self.entry2id_dict = defaultdict()
+        self.id2entry_dict = defaultdict()
+
+        if use_pad:
+            self.add_entry(self.PAD_ENTRY)
+        if use_unk:
+            self.add_entry(self.UNK_ENTRY)
+            self.id2entry_dict.default_factory = \
+                lambda: self.UNK_ENTRY
+            self.entry2id_dict.default_factory = \
+                lambda: self.entry2id_dict[self.UNK_ENTRY]
+
+        assert method in ("indexing", "one-hot")
+        self.method = method
+        self.use_pad = use_pad
+        self.use_unk = use_unk
+
+    def get_pad_id(self):
+        if self.use_pad:
+            return self.entry2id(self.PAD_ENTRY)
+        return None
+
+    def get_unk_id(self):
+        if self.use_unk:
+            return self.entry2id(self.UNK_ENTRY)
+        return None
+
+    def add_entry(self, entry: Any):
```

Comment:
Maybe just call this function `add`. Entry is a bit abused in the Forte terminology.

Reply:
Got it.